### PR TITLE
API-diff between 7.0-preview2 and 7.0-preview3

### PR DESCRIPTION
**Pull request by carlossanlop**
The bugs we found in the API diffs for Preview1 and Preview2 were fixed here: https://github.com/dotnet/arcade/pull/9172

ASP.NET: @dotnet/aspnet-api-review 

WinForms: @merriemcgaw @RussKie 

WPF: @dreddy-work @RussKie 

Libraries: @danmoseley @jeffhandley @karelz @ericstj @stephentoub 

@dotnet/area-extensions-filesystem 
@dotnet/area-extensions-hosting 
@dotnet/area-extensions-logging 
@dotnet/area-extensions-primitives 
@dotnet/area-system-runtime 
@dotnet/area-system-numerics 
@dotnet/area-system-diagnostics-activity 
System.Net: @dotnet/ncl 
@dotnet/interop-contrib
@dotnet/area-system-security 
@dotnet/area-system-text-regularexpressions

### CODE REVIEW

**Comment by tarekgh** [comment]
I am seeing something strange in the system namespace. here is some example:

```
-        public static DateOnly MaxValue { get; }
+        static DateOnly MaxValue { get; }
```

Is this intentional?

**Comment by tarekgh** [comment]
I am not seeing `System.Composition.Hosting` is included. Here is the new exposed APIs https://github.com/dotnet/core/issues/7108#issuecomment-1069723651. Or we are not including the OOB libraries?

**Comment by tarekgh** [comment]
The following System.Diagnostics APIs LGTM.

```diff
     public class Activity : IDisposable {
+        public bool HasRemoteParent { get; }
     }
     public readonly struct ActivityCreationOptions<T> {
+        public string? TraceState { get; set; }
     }
```

**Review by dreddy-work — APPROVED** [review]
`Microsoft.WindowsDesktop.App` APIs look good to me.

**Review by tannergooding — COMMENTED** [review]
Runtime changes LGTM

**Review by maryamariyan — APPROVED** [review]
M.E.Logging/Caching/Configuration/Primitives diff LGTM.

cc @halter73 for `Hosting`

**Comment by AaronRobinsonMSFT** [comment]
`System.Runtime.InteropServices` LGTM.

File: release-notes/7.0/preview/api-diff/preview3/Microsoft.AspNetCore.App/7.0-preview3_Microsoft.Extensions.Hosting.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
++        public string ApplicationName { get; set; }
++        public string[] Args { get; set; }
++        public ConfigurationManager Configuration { get; set; }
++        public string ContentRootPath { get; set; }
++        public bool DisableDefaults { get; set; }
++        public string EnvironmentName { get; set; }
++    }
+     public interface IHostEnvironment {
+-        string ApplicationName { get; set; }
++        string? ApplicationName { get; set; }
```
Comment (halter73) [inline release-notes/7.0/preview/api-diff/preview3/Microsoft.AspNetCore.App/7.0-preview3_Microsoft.Extensions.Hosting.md:38]: Just a heads up. I think this should be non-nullable, but we do need to figure out what this gets set to when `Assembly.GetEntryAssembly()` returns null. I think this is fine for previews, but we should try to fix this for the final release of .NET 7. I filed an issue: #68512.

**Review by halter73 — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Comment by carlossanlop** [comment]
> I am not seeing `System.Composition.Hosting` is included. Here is the new exposed APIs [#7108 (comment)](https://github.com/dotnet/core/issues/7108#issuecomment-1069723651). Or we are not including the OOB libraries?

That's my understanding, @tarekgh - We do not include OOBs in this report.

**Review by bartonjs — APPROVED** [review]
Crypto LGTM

**Pull request merged**